File: app/src/main/java/com/krystian/chessclock/CustomMatchDialogFragment.java
package com.krystian.chessclock;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;
import android.widget.SeekBar;
import android.widget.TextView;
import android.widget.Toast;

public class CustomMatchDialogFragment extends DialogFragment implements SeekBar.OnSeekBarChangeListener {

    private View dialogView;

    private SeekBar numberOfGamesBar;
    private TextView numberOfGamesText;
    private EditText matchName;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final AlertDialog.Builder builder= new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = requireActivity().getLayoutInflater();
        dialogView = inflater.inflate(R.layout.custom_match_dialog, null);
        setViewComponents();
        builder.setView(dialogView)
                .setTitle(R.string.add_new_match)
                .setPositiveButton(R.string.ok_button, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int id) {
                        String name = matchName.getText().toString();
                        if(name.isEmpty())
                            Toast.makeText(getActivity(), R.string.no_name_chosen, Toast.LENGTH_SHORT).show();
                        else {
                            Intent intent = new Intent(getContext(), CustomMatchActivityList.class);
                            intent.putExtra("matchName", name);
                            intent.putExtra("numberOfGames", numberOfGamesBar.getProgress()+2);
                            startActivity(intent);
                            Toast.makeText(getActivity(), R.string.match_created, Toast.LENGTH_SHORT).show();
                        }
                    }
                })
                .setNegativeButton(R.string.cancel_button, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {}
                });

        return builder.create();
    }

    public void setViewComponents() {
        numberOfGamesBar = dialogView.findViewById(R.id.number_of_games_bar);
        numberOfGamesText = dialogView.findViewById(R.id.number_of_games_dialog);
        matchName = dialogView.findViewById(R.id.match_name);
        numberOfGamesBar.setOnSeekBarChangeListener(this);

        numberOfGamesBar.setProgress(3);
        numberOfGamesText.setText(getString(R.string.number_of_games, numberOfGamesBar.getProgress()+2));
    }


    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
        int numberOfGames = progress + 2; //range 2-30
        numberOfGamesText.setText(getString(R.string.number_of_games, numberOfGames));
    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {

    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {

    }
}

File: app/src/main/java/com/krystian/chessclock/dialogs/CustomMatchDialogFragment.kt
package com.krystian.chessclock.dialogs

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.core.widget.doAfterTextChanged
import androidx.fragment.app.DialogFragment
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.krystianrymonlipinski.chessclock.R
import com.krystianrymonlipinski.chessclock.databinding.DialogFragmentNewCustomMatchBinding


class CustomMatchDialogFragment(
    private val callback: Callback
) : DialogFragment() {

    private val _binding: DialogFragmentNewCustomMatchBinding by lazy {
        DialogFragmentNewCustomMatchBinding.inflate(layoutInflater)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return MaterialAlertDialogBuilder(requireContext())
            .setTitle(R.string.add_new_match)
            .setView(_binding.root)
            .setPositiveButton(R.string.ok_button) { dialog: DialogInterface?, _: Int ->
                callback.onNewMatchCreated(
                    _binding.etMatchName.text.toString(),
                    _binding.slNumberOfGames.value.toInt()
                )
                dialog?.dismiss()
            }
            .setNegativeButton(R.string.cancel_button) { dialog: DialogInterface?, _: Int ->
                dialog?.dismiss()
            }
            .show()
            .also {
                setGamesText(_binding.slNumberOfGames.value.toInt())
                setIsPositiveButtonEnabled(it, _binding.tfMatchName.editText?.text?.isNotBlank() ?: false)
                setupListeners(it)
            }
    }

    private fun setupListeners(alertDialog: AlertDialog) {
        with(_binding) {
            slNumberOfGames.addOnChangeListener { _, value, _ ->
                setGamesText(value.toInt())
            }
            tfMatchName.editText?.doAfterTextChanged { text ->
                text?.let {
                    setIsPositiveButtonEnabled(alertDialog, text.isNotBlank())
                }
            }
        }
    }

    private fun setGamesText(number: Int) {
        _binding.tvNumberOfGames.text = String.format(getString(R.string.number_of_games, number))
    }

    private fun setIsPositiveButtonEnabled(alertDialog: AlertDialog, isEnabled: Boolean) {
        alertDialog.getButton(DialogInterface.BUTTON_POSITIVE)?.isEnabled = isEnabled
    }

    interface Callback {
        fun onNewMatchCreated(name: String, numberOfGames: Int)
    }
}
